check-rabbitmq-exchange/lib/Nagios/Check/RabbitMQ/Process.hs

```diff
@@ -7,22 +7,30 @@ import           Control.Monad
 import           Data.Aeson
 import           Data.ByteString.Lazy.Char8  (ByteString)
 import qualified Data.ByteString.Lazy.Char8  as BSL
+import           Data.Int
 import           Data.Maybe
 import qualified Data.Text                   as T
+import qualified Data.Vector                 as V
 import           Nagios.Check.RabbitMQ.Types
 import           System.Nagios.Plugin
 
-checkRawExchange :: ByteString -> CheckOptions -> IO ()
-checkRawExchange bs opts = case eitherDecode bs of
-    Left  e -> runNagiosPlugin $ addResult Unknown $ T.pack ( "Decode failed with: " ++ e )
-    Right x -> checkExchange x opts
+checkRawExchange :: ByteString -> CheckOptions -> Int64 -> IO ()
+checkRawExchange bs opts connCount = case eitherDecode bs of
+    Left  e -> runNagiosPlugin $ addResult Unknown $ T.pack ( "Exchange decode failed with: " ++ e )
```

Comment:
- If we're making the message more specific (good idea), can we go further and also specify what we failed to decode (what type of message we're expecting)?

 - Why not run this in the NagiosPlugin monad rather than IO? That way we don't have to call `runNagiosPlugin` twice, and any unexpected fatal exceptions in the first half of the function will cause the check to successfully report an error rather than crash and burn.